### PR TITLE
Upstream VSCode Python changes from v2024.14.1 to v2024.20.0

### PR DESCRIPTION
**Pull request by seeM**
Addresses #5567.

I also bumped our kernel dependencies (jedi, jedi-language-server, etc) which required updates to the patch files.

### QA Notes

Build and run Positron locally and check that things generally look okay. I'll also run the full test suite GH action.

### CODE REVIEW

**Comment by wesm** [comment]
Built locally on Linux (Fedora 40) and kicked the tires and things look good. Did you check out a release build?

File: extensions/positron-python/package.json [hunk 1]
```diff
@@ -753,9 +785,16 @@
                     "scope": "resource",
                     "type": "boolean",
                     "tags": [
-                        "experimental"
+                        "onExP",
+                        "preview"
                     ]
                 },
+                "python.REPL.provideVariables": {
```
Comment (isabelizimm) [inline extensions/positron-python/package.json:792]: should we leave out/hide the native REPL settings as we expect people to use Positron's console?

**Review by isabelizimm — APPROVED** [review]
looks good playing around with things locally!

**Comment by seeM** [comment]
I've also built a release locally and it's working as expected.

**Pull request merged**